municipal_finance/resources.py

```diff
@@ -0,0 +1,68 @@
+from import_export import resources
+
+from .models import (
+    AmountTypeV2,
+    CflowItemsV2,
+    GovernmentFunctionsV2,
+    IncexpItemsV2,
+    BsheetItemsV1,
+    BsheetItemsV2,
+    BsheetFactsV1,
+    BsheetFactsV2,
+)
+
+
+class AmountTypeV2Resource(resources.ModelResource):
+    class Meta:
+        model = AmountTypeV2
+        import_id_fields = ['code']
+
+
+class CashflowItemsV2Resource(resources.ModelResource):
+    class Meta:
+        model = CflowItemsV2
+        import_id_fields = ['code']
+
+
+class GovernmentFunctionsV2Resource(resources.ModelResource):
+    class Meta:
+        model = GovernmentFunctionsV2
+        import_id_fields = ['code']
+
+
+class IncexpItemsV2Resource(resources.ModelResource):
+    class Meta:
+        model = IncexpItemsV2
+        import_id_fields = ['code']
+
+
+class BsheetItemsV1Resource(resources.ModelResource):
+    class Meta:
+        model = BsheetItemsV1
+        import_id_fields = ['code']
+
+
+class FinancialPositionItemsV2Resource(resources.ModelResource):
+    class Meta:
+        model = BsheetItemsV2
+        import_id_fields = ['code']
+
+
+class BsheetFactsV2Resource(resources.ModelResource):
+    class Meta:
+        model = BsheetFactsV2
+        import_id_fields = [
+            'demarcation_code',
+            'period_code',
+            'item',
+        ]
+
+
+class BsheetFactsV1Resource(resources.ModelResource):
+    class Meta:
+        model = BsheetFactsV1
+        import_id_fields = [
+            'demarcation_code',
+            'period_code',
+            'item_code',
+        ]
```

Comment:
This is not something that should accidentally be used in production, because import_id_fields does not guarantee that the whole "import set" is deleted - so a mistaken row in an old upload won't be removed when that period is re-uploaded with a correction.

I don't think something that might accidentally be used incorrectly should be in prod code - perhaps we can move it to test code where we can rely on the test framework to flush the DB between tests.

Reply:
As far as I understand what you're saying, and how resources import works, the 'mistaken' rows will indeed be replaced, as long as the demarcation_code, period_code and item_code are the same.

Also keep in mind that these can only be used to import data that was directly exported from our database, and not datasets provided by National Treasury.

These classes also don't show anything on admin yet. Just in case you though that might be the case.

I get what you're saying about keeping things that are only used by tests with tests, and I'll move more in that direction going forward.

Can we leave it here for now?